brainiak/eventseg/event.py

```diff
@@ -0,0 +1,363 @@
+#  Copyright 2016 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Event segmentation using a Hidden Markov Model
+
+Given an ROI timeseries, this class uses an annealed fitting procedure to
+segment the timeseries into events with stable activity patterns. After
+learning the signature activity pattern of each event, the model can then be
+applied to other datasets to identify a corresponding sequence of events.
+"""
+
+# Authors: Chris Baldassano and Cătălin Iordan (Princeton University)
+
+import numpy as np
+from scipy import stats
+import logging
+import copy
+from sklearn.base import BaseEstimator
+from sklearn.utils.validation import check_is_fitted, check_array
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "EventSegment",
+]
+
+
+class EventSegment(BaseEstimator):
+    """Class for learning and representing an event segmentation of
+    continuous fMRI data
```

Comment:
Docstrings should start with a one-line summary:
https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings